### PR TITLE
feat: add support for service account impersonation

### LINKED ISSUE #96: Add support for service account impersonation
### Feature Description

This is for adapting/porting https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/pull/1426.

### Sample code

_No response_

### Alternatives Considered

_No response_

### Additional Details

_No response_

### PR DESCRIPTION
**Pull request by ttosta-google**
Fixes #96

### CODE REVIEW

File: .envrc.example [hunk 1]
```diff
@@ -2,3 +2,6 @@ export ALLOYDB_DB="some-db"
 export ALLOYDB_USER="some-user"
 export ALLOYDB_PASS="some-password"
 export ALLOYDB_INSTANCE_NAME="projects/<PROJECT>/locations/<REGION>/clusters/<CLUSTER>/instances/<INSTANCE>"
+export ALLOYDB_INSTANCE_IP="some-IP-address"
+export ALLOYDB_IAM_USER="some-IAM-user"
```
Comment (enocom) [inline .envrc.example:6]: Let's make this some-user@my-project.iam. Same below, just so folks know the format.

File: README.md [hunk 2]
```diff
@@ -157,6 +157,76 @@ performance from a connection pool.
...
+Token Creator role on serviceAccountB. Then serviceAccountB must have the
+Token Creator role on serviceAccountC. Finally, serviceAccountC must have the
+Token Creator role on `alloydbTargetPrincipal`. If unset, the IAM principal
+must have the Token Creator role on `alloydbTargetPrincipal`.
+
+```java
+config.addDataSourceProperty("alloydbTargetPrincipal",
+    "TARGET_SERVICE_ACCOUNT");
+config.addDataSourceProperty("alloydbDelegates", 
+    "SERVICE_ACCOUNT_1, SERVICE_ACCOUNT_2");
```
Comment (enocom) [inline README.md:223]: I think these shouldn't be separate with the space, just a comma. Is that right?

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java [hunk 3]
```diff
@@ -84,6 +85,11 @@ public ConnectionInfo getConnectionInfo(InstanceName instanceName, KeyPair keyPa
         info.getIpAddress(), info.getInstanceUid(), clientCertificate, certificateChain);
   }
 
+  @Override
+  public void close() {
```
Comment (enocom) [inline alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java:89]: Does this need to be public?

**Review by enocom — APPROVED** [review]
The design here looks great. Thanks for thinking through all the options.

Some minor nits, but I'll approve now.

File: alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java (hunk 3 above)
Comment (ttosta-google) [reply alloydb-jdbc-connector/src/main/java/com/google/cloud/alloydb/DefaultConnectionInfoRepository.java:89]: It does. If we remove `public` the error below is displayed: 

`Cannot reduce the visibility of the inherited method from CloseableJava(67109273)`

File: README.md (hunk 2 above)
Comment (ttosta-google) [reply README.md]: Right. Updated.

File: .envrc.example (hunk 1 above)
Comment (ttosta-google) [reply .envrc.example]: Updated.

**Pull request merged**